mmv1/products/discoveryengine/SearchEngine.yaml

```diff
@@ -0,0 +1,153 @@
+    resource_inside_response: true
+
+import_format:
+  [
+    'projects/{{project}}/locations/{{location}}/collections/{{collection_id}}/engines/{{engine_id}}',
+  ]
+custom_code: !ruby/object:Provider::Terraform::CustomCode
+  encoder: templates/terraform/encoders/discovery_engine_search_engine_hardcode_solution_type.go.erb
+examples:
+  - !ruby/object:Provider::Terraform::Examples
+    name: "discoveryengine_searchengine_basic"
+    primary_resource_id: 'basic'
+    primary_resource_name:
+      'fmt.Sprintf("tf_test_search_engine%s", context["random_suffix"])'
+    vars:
+      engine_id: "example-engine-id"
+      data_store_id: "example-datastore-id"
+
+parameters:
+  - !ruby/object:Api::Type::String
+    name: 'engine_id'
+    description: |
+      Unique ID to use for Search Engine App.
+    required: true
+    immutable: true
+    url_param_only: true
+  - !ruby/object:Api::Type::String
+    name: 'collection_id'
+    description: |
+      The collection ID.
+    required: true
+    immutable: true
+    url_param_only: true
+  - !ruby/object:Api::Type::String
+    name: 'location'
+    description: |
+      Location.
+    required: true
+    immutable: true
+    url_param_only: true
+
+properties:
+  - !ruby/object:Api::Type::String
+    name: 'name'
+    description: |
+      The unique full resource name of the search engine. Values are of the format
+      `projects/{project}/locations/{location}/collections/{collection_id}/engines/{engine_id}`.
+      This field must be a UTF-8 encoded string with a length limit of 1024
+      characters.
+    output: true
+  - !ruby/object:Api::Type::Enum
+    name: 'industryVertical'
+    description: |
+      The industry vertical that the engine registers. The restriction of the Engine industry vertical is based on DataStore: If unspecified, default to GENERIC. Vertical on Engine has to match vertical of the DataStore liniked to the engine.
+    immutable: true
+    values:
+      - :GENERIC
+      - :MEDIA
+  - !ruby/object:Api::Type::String
+    name: 'displayName'
+    description: |
+      Required. The display name of the engine. Should be human readable. UTF-8 encoded string with limit of 1024 characters.
+    required: true
+  - !ruby/object:Api::Type::Array
+    name: 'dataStoreIds'
+    description: |
+      The data stores associated with this engine. For SOLUTION_TYPE_SEARCH type of engines, they can only associate with at most one data store.
+    required: true
+    immutable: true
+    item_type: Api::Type::String
+  - !ruby/object:Api::Type::Time
+    name: 'createTime'
+    description: |
+      Timestamp the Engine was created at.
+    output: true
+  - !ruby/object:Api::Type::Time
+    name: 'updateTime'
+    description: |
+      Timestamp the Engine was last updated.
+    output: true
+  - !ruby/object:Api::Type::NestedObject
+    name: 'searchEngineConfig'
+    description: |
+      Configurations for a Search Engine.
+    default_from_api: true
```

Comment:
Should we remove this?

Reply:
Are you talking about default_from_api?

Reply:
Yes I'm not very sure, but should we remove default_from_api?

Reply:
removed